compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEBufferizeToAllocation.cpp

```diff
@@ -45,19 +45,31 @@ static LogicalResult applyBufferizeToAllocation(RewriterBase &rewriter,
 /// elementwise op's input is the output of the matmul op and has already been
 /// promoted, there is no need to promote such operand again.
 static SmallVector<Value> getInputOutputOperands(linalg::LinalgOp &linalgOp) {
-  SmallVector<Value> operands;
-  for (auto operand : linalgOp->getOperands()) {
-    if (isMatmulProducerOfElementwise(linalgOp)) {
-      auto defOp = operand.getDefiningOp<linalg::LinalgOp>();
-      // Continue the loop without including the operand which is the output of
-      // matmul.
-      if (defOp && isMatmul(defOp)) {
-        continue;
-      }
+  if (isMatmulProducerOfElementwise(linalgOp)) {
+    SmallVector<Value> operands;
+    for (auto operand : linalgOp->getOperands()) {
+      if (isMatmulInDefChain(operand)) continue;
+      operands.push_back(operand);
     }
-    operands.push_back(operand);
+    return operands;
   }
-  return operands;
+  return linalgOp->getOperands();
+}
+
+/// Utility to fetch input operands from the LinalgOp (matmul or elementwise
+/// op). For matmul-elementwise special case, since one of the elementwise op's
+/// input is the output of the matmul op and has already been promoted, there is
+/// no need to promote such operand again.
+static SmallVector<Value> getInputOperands(linalg::LinalgOp &linalgOp) {
+  if (isMatmulProducerOfElementwise(linalgOp)) {
+    SmallVector<Value> operands;
+    for (auto operand : linalgOp.getDpsInputs()) {
```

Comment:
`auto` -> `Value`.

Expand `auto` here and elsewhere.

Reply:
Just wondering is there a particular reason to do so?

Reply:
Thanks for addressing the comment.

Regarding the reason - yes, it increases the verbosity of the code.

`auto` is to be used only when the data type of the LHS is known within the one-line context. Example of cases where we're to use `auto` is `dyn_cast<....>`, `cast<....>`, `rewrite.create<...>`, etc.

In the example of cases like above the reader of the code doesn't know whether `getDpsInputs()` is returning a `Value`, `OpOperand` or some other data type. So, it helps in stating the LHS' data type in cases like these and use `auto` in cases where the RHS explicitly mentions what the data type is.